### ISSUE TITLE
Inconsistent errors between one-to-many and many-to-many associations

### ISSUE THREAD

**Issue by kotarou3** [balderdashy/sails#3046]
Adding a duplicate child to a one-to-many association silently "fails" (nothing is changed, other than update time), while for a many-to-many association, an error is thrown.

```sh
$ sails -v
0.11.0
```

`api/models/A.js`:
```js
"use strict";

module.exports = {
    attributes: {
        "one-a-to-many-b": {
            collection: "B",
            via: "one-a-to-many-b"
        },
        "many-a-to-many-b": {
            collection: "B",
            via: "many-a-to-many-b"
        }
    }
};
```

`api/models/B.js`:
```js
"use strict";

module.exports = {
    attributes: {
        "one-a-to-many-b": {
            model: "A"
        },
        "many-a-to-many-b": {
            collection: "A",
            via: "many-a-to-many-b"
        }
    }
};
```

In `sails console`:
```sh
sails> A.create({}).exec(console.log)
null { createdAt: '2015-06-29T14:17:30.435Z',
  updatedAt: '2015-06-29T14:17:30.435Z',
  id: 1 }

sails> B.create({"one-a-to-many-b": 1, "many-a-to-many-b": [1]}).exec(console.log)
null { 'one-a-to-many-b': 1,
  createdAt: '2015-06-29T14:18:36.063Z',
  updatedAt: '2015-06-29T14:18:36.063Z',
  id: 1 }

sails> A.find().populate("one-a-to-many-b").populate("many-a-to-many-b").exec(console.log)
null [ { 'one-a-to-many-b': 
     [ { 'one-a-to-many-b': 1,
         createdAt: '2015-06-29T14:18:36.063Z',
         updatedAt: '2015-06-29T14:18:36.071Z',
         id: 1 } ],
    'many-a-to-many-b': 
     [ { 'one-a-to-many-b': 1,
         createdAt: '2015-06-29T14:18:36.063Z',
         updatedAt: '2015-06-29T14:18:36.071Z',
         id: 1 } ],
    createdAt: '2015-06-29T14:17:30.435Z',
    updatedAt: '2015-06-29T14:17:30.435Z',
    id: 1 } ]

sails> A.find().populate("one-a-to-many-b").populate("many-a-to-many-b").exec(function (e, a) {
... a[0]["one-a-to-many-b"].add(1);
... a[0].save(console.log);
... })
null { 'one-a-to-many-b': 
   [ { 'one-a-to-many-b': 1,
       createdAt: '2015-06-29T14:18:36.063Z',
       updatedAt: '2015-06-29T14:20:44.367Z',
       id: 1 } ],
  'many-a-to-many-b': 
   [ { 'one-a-to-many-b': 1,
       createdAt: '2015-06-29T14:18:36.063Z',
       updatedAt: '2015-06-29T14:20:44.367Z',
       id: 1 } ],
  createdAt: '2015-06-29T14:17:30.435Z',
  updatedAt: '2015-06-29T14:20:44.356Z',
  id: 1 }

sails> A.find().populate("one-a-to-many-b").populate("many-a-to-many-b").exec(function (e, a) {
... a[0]["many-a-to-many-b"].add(1);
... a[0].save(console.log);
... })
[ { type: 'insert',
    collection: 'a_many-a-to-many-b__b_many-a-to-many-b',
    criteria: { 'b_many-a-to-many-b': 1, 'a_many-a-to-many-b': 1 },
    values: { 'b_many-a-to-many-b': 1, 'a_many-a-to-many-b': 1 },
    err: [Error: Trying to '.add()' an instance which already exists!] } ]
```

**Comment by kotarou3** [comment]
Also, not really related to the issue, but the fact that the duplicate error is an `Array` rather than an `Error` is a bit strange to me. Missing a stack trace could potentially make it hard to debug it.

**Comment by sailsbot** [comment]
Thanks for posting, @kotarou3.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
